[refactor] Modify the README to apply the new installation method (#239)
apache/pulsar-client-node: README.md

The Pulsar Node.js client can be used to create Pulsar producers and consumers in Node.js.

This library works only in Node.js 10.x or later because it uses the
[node-addon-api](https://github.com/nodejs/node-addon-api) module to wrap the C++ library.

## How to install

> **Note**
>

> > Only available for versions after 1.8.0. For versions before 1.8.0, you need to install the C++ client first, and switch to the corresponding version branch to view the specific steps.

Reviewer: ```suggestion
> These instructions are only available for versions after 1.8.0. For versions previous to 1.8.0, you need to install the C++ client first. Please switch to the corresponding version branch of this repo to read the specific instructions.
```

> 1. You can use `npm` or `yarn` to install `pulsar-client-node`

Reviewer: Now we don't need to use multiple steps. I suggest keeping only the install instructions:

````diff
diff --git a/README.md.old b/README.md
index 2f781c9..fbb4d2f 100644
--- a/README.md.old
+++ b/README.md
@@ -32,18 +32,19 @@ This library works only in Node.js 10.x or later because it 
uses the
 >
 > Only available for versions after 1.8.0. For versions before 1.8.0, you need 
to install the C++ client first, and switch to the corresponding version branch 
to view the specific steps.
 
-1. You can use `npm` or `yarn` to install `pulsar-client-node`
+### `npm`
 
 ```shell
 npm install pulsar-client
-#or
-yarn add pulsar-client
 ```
 
-2. Run Sample code
+### `yarn`
 
-Please refer to [examples](https://github.com/apache/pulsar-client-node/tree/master/examples).
+```shell
+yarn add pulsar-client
+```
 
+After install, you can run the [examples](https://github.com/apache/pulsar-client-node/tree/master/examples).
 
 ### Prebuilt binaries
````

Author: oh, Let us only keep `npm`.

```shell
npm install pulsar-client
#or
yarn add pulsar-client
```

2. Run Sample code

Please refer to [examples](https://github.com/apache/pulsar-client-node/tree/master/examples).

> ### Prebuilt binaries

Reviewer: Why do we write this section? Shall we move it to the "How to build" chapter?

Author: This is a simple introduction to the principle, and I should put it before `How to install` section.

The module uses [node-pre-gyp](https://github.com/mapbox/node-pre-gyp) to download the prebuilt binary for your platform, if it exists. 
These binaries are hosted on ASF dist subversion. The following targets are currently provided:

Format: `napi-{platform}-{libc}-{arch}`
- napi-darwin-unknown-x64.tar.gz
- napi-linux-glibc-arm64.tar.gz
- napi-linux-glibc-x64.tar.gz
- napi-linux-musl-arm64.tar.gz
- napi-linux-musl-x64.tar.gz
- napi-win32-unknown-ia32.tar.gz
- napi-win32-unknown-x64.tar.gz

`darwin-arm64` systems are not currently supported, you can refer `How to build` to build from source.

## How to build

> To build from source, you need to install the CPP client first.

Reviewer: ```suggestion
```

Already reflected in step 2. If you write this line followed "clone repository", it reads confusing.

Author: agree, I removed this line.

1. Clone repository.
```shell

> $ git clone https://github.com/apache/pulsar-client-node.git

Reviewer: Keep consistency not to use `$`.

Reviewer: ```suggestion
git clone https://github.com/apache/pulsar-client-node.git
```

ditto others.

$ cd pulsar-client-node
```

> 2. Install CPP client.

Reviewer: ```suggestion
2. Install C++ client.
```

Keep consistency to refer `C++`.

> Select the appropriate installation method from below depending on your operating system.

Reviewer: ```suggestion
Select the appropriate installation method from below depending on your operating system:
```

> Install c++ client on mac

Reviewer: ```suggestion
Install C++ client on macOS:
```

```shell
$ pkg/mac/build-cpp-deps-lib.sh
$ pkg/mac/build-cpp-lib.sh
```

> Install c++ client on Linux

Reviewer: ```suggestion
Install C++ client on Linux:
```

```shell
$ build-support/install-cpp-client.sh
```

> Install c++ client on Windows(Need install curl and 7z first)

Reviewer: ```suggestion
Install C++ client on Windows (required preinstall `curl` and `7z`):
```

```shell
$ pkg\windows\download-cpp-client.bat
```

> 3. Build NAPI from source.

Reviewer: I may suggest you use `### 3. Build NAPI from source` and ditto other steps, since step 2 is relatively complex.

Author: adopt

```shell
npm install --build-from-source 
```

## Documentation

> * Please see https://pulsar.apache.org/docs/en/client-libraries-node for more details about the Pulsar Node.js client.  

Reviewer: ```suggestion
* Please see https://pulsar.apache.org/docs/client-libraries-node/ for more details about the Pulsar Node.js client.  
```

Use a concise URL.
